addons/web/static/src/views/relational_model/group.js

```diff
@@ -56,6 +54,20 @@ export class Group extends DataPoint {
     // Public
     // -------------------------------------------------------------------------
 
+    applyFilter(filter) {
+        this.model._updateConfig(this.config, { groupFilterDomain: filter }, { noReload: true });
+        if (this.config.groupFilterDomain) {
+            return this.list.load({
+                domain: Domain.and([
+                    this.config.initialDomain,
+                    this.config.groupFilterDomain,
+                ]).toList(),
+            });
+        } else {
+            return this.list.load({ domain: this.config.initialDomain });
+        }
```

Comment:
here you change the config on t_0, and then load the data which arrive in t_1, meaning that in between the state is inconsistent